### PR TITLE
Fix trappy timeouts in `o.e.a.a.cluster.*`

### PR DESCRIPTION
**Pull request by DaveCTurner**
Removes all usages of `TRAPPY_IMPLICIT_DEFAULT_MASTER_NODE_TIMEOUT` in
cluster-related APIs in `:server`.

Relates #107984

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java [hunk 1]
```diff
@@ -666,15 +666,15 @@ public void testRestoreShrinkIndex() throws Exception {
         assertAcked(indicesAdmin().prepareDelete(sourceIdx).get());
         assertAcked(indicesAdmin().prepareDelete(shrunkIdx).get());
         internalCluster().stopRandomDataNode();
-        clusterAdmin().prepareHealth().setTimeout(TimeValue.timeValueSeconds(30)).setWaitForNodes("1");
+        clusterAdmin().prepareHealth(TEST_REQUEST_TIMEOUT).setTimeout(TimeValue.timeValueSeconds(30)).setWaitForNodes("1");
```
Comment (nik9000) [inline server/src/internalClusterTest/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java:669]: More timeouts. Still ok?

File: x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/action/TransportMonitoringMigrateAlertsAction.java [hunk 2]
```diff
@@ -98,7 +98,11 @@ protected void masterOperation(
             Settings.Builder decommissionAlertSetting = Settings.builder().put(Monitoring.MIGRATION_DECOMMISSION_ALERTS.getKey(), true);
             client.admin()
                 .cluster()
-                .prepareUpdateSettings()
+                .prepareUpdateSettings(
+                    request.masterNodeTimeout(),
+                    /* TODO expose separate ack timeout? use masterNodeTimeout() for now */
```
Comment (nik9000) [inline x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/action/TransportMonitoringMigrateAlertsAction.java:103]: Do you want to start a meta issue with links to all of these potential follow-up items? I get the sense this PR isn't going to *solve* these, but it does identify them.

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/TransformInternalIndex.java [hunk 3]
```diff
@@ -389,7 +390,10 @@ protected static boolean allPrimaryShardsActiveForLatestVersionedIndex(ClusterSt
     }
 
     private static void waitForLatestVersionedIndexShardsActive(Client client, ActionListener<Void> listener) {
-        ClusterHealthRequest request = new ClusterHealthRequest(TransformInternalIndexConstants.LATEST_INDEX_VERSIONED_NAME)
+        ClusterHealthRequest request = new ClusterHealthRequest(
+            TimeValue.THIRTY_SECONDS /* TODO should this be longer/configurable? */,
```
Comment (nik9000) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/TransformInternalIndex.java:394]: Another one that might want to be in the meta issue.

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java [hunk 4]
```diff
@@ -684,7 +685,10 @@ private void setIndexReadOnly(ActionListener<AcknowledgedResponse> listener) {
     }
 
     private void waitForIndexGreen(ActionListener<ClusterHealthResponse> listener) {
-        ClusterHealthRequest request = new ClusterHealthRequest(enrichIndexName).waitForGreenStatus();
+        ClusterHealthRequest request = new ClusterHealthRequest(
+            TimeValue.THIRTY_SECONDS /* TODO should this be longer/configurable? */ ,
```
Comment (nik9000) [inline x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java:689]: Should this one be in a meta issue?

**Review by nik9000 — APPROVED** [review]
I'm sure I'm not the right one to look at this, but the codeowners file pinged me. So here I am. I've scanned the change and it:
1. Adds timeouts to the ctors on many requests.
2. In tests passes `TEST_REQUEST_TIMEOUT` to it.
3. In a few production files it passes THIRTY_SECONDS with a note to make it configurable.
4. In the remaining files it plugs a master_node_timeout that's been sent of HTTP or has been added to request that's spawning it.


That's pretty much what it does. 350 times.

**Review by idegtiarenko — APPROVED** [review]

File: server/src/internalClusterTest/java/org/elasticsearch/cluster/ClusterHealthIT.java [hunk 5]
```diff
@@ -74,13 +77,16 @@ public void testHealth() {
         createIndex("test1");
 
         logger.info("--> running cluster health on an index that does exists");
-        healthResponse = clusterAdmin().prepareHealth("test1").setWaitForGreenStatus().setTimeout(TimeValue.timeValueSeconds(10)).get();
+        healthResponse = clusterAdmin().prepareHealth(TEST_REQUEST_TIMEOUT, "test1")
+            .setWaitForGreenStatus()
+            .setTimeout(TimeValue.timeValueSeconds(10))
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply server/src/internalClusterTest/java/org/elasticsearch/cluster/ClusterHealthIT.java:82]: No, this is the timeout to wait for `green` after actually reaching the master, whereas `masterNodeTimeout` is largely about how long it takes to get the message to the master in the first place.

File: server/src/internalClusterTest/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/internalClusterTest/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java:669]: Yep.

File: x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/action/TransportMonitoringMigrateAlertsAction.java (hunk 2 above)
Comment (DaveCTurner) [reply x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/action/TransportMonitoringMigrateAlertsAction.java:103]: The point of https://github.com/elastic/elasticsearch/issues/107984 is to make these choices more explicit in the calling code, and hopefully to reduce the rate at which we add new things like this, but I expect a meta-issue to track any followups would just sit unaddressed for years anyway. It's just not very important. We've had these fixed (or at least confused) timeouts for a long time.

**Comment by DaveCTurner** [comment]
> That's pretty much what it does. 350 times.

The key improvement made here is to remove 7 references to `MasterNodeRequest#TRAPPY_IMPLICIT_DEFAULT_MASTER_NODE_TIMEOUT`. But yeah there's a lot of noise to plumb in more appropriate values indeed.

**Comment by nik9000** [comment]
> The key improvement made here is to remove 7 references to `MasterNodeRequest#TRAPPY_IMPLICIT_DEFAULT_MASTER_NODE_TIMEOUT`. But yeah there's a lot of noise to plumb in more appropriate values indeed.

Right. you're forcing the compiler to make  you make a choice. 330 of those choices are "the test timeone is fine" and that's great. But it's important to catch the 20 times when that isn't fine and the compiler helps here. : +1:

**Pull request merged**